src/werkzeug/datastructures.py

```diff
@@ -2952,6 +2952,11 @@ def save(self, dst, buffer_size=16384):
         from shutil import copyfileobj
 
         close_dst = False
+        if not PY2:
+            from pathlib import PurePath
+
+            if isinstance(dst, PurePath):
```

Comment:
for correct support of pathlib there  is a need to actually use `path.open` if possible/sensible

for basic support of "path-like" objects, recent python3 (aka 3.6 plus also support passing it to open)

a basic acceptance test would help to guide what to keep in place later on